addons/web/static/src/core/notebook/notebook.js

```diff
@@ -39,9 +69,14 @@ export class Notebook extends Component {
             },
             () => [this.state.currentPage]
         );
-        onWillUpdateProps(
-            (nextProps) => (this.state.currentPage = this.computeActivePage(nextProps))
-        );
+        onWillUpdateProps((nextProps) => {
+            this.computePages(nextProps);
```

Comment:
this seems like a bad idea. the set of pages should not be recomputed each time, in my opinion.
Also, the next line is wrong: we should not reset the currentpage at every render. This may cause issues such as a random unrelated rendering changing the current page.

Reply:
yep, I see, This seems scary. what happens when you remove the currentpage recompute?

Reply:
Some FormView tests fail: when changing tab visibility, missing tabs/content, etc.